tests/regionizers/test_s2_regionizer.py

```diff
@@ -0,0 +1,58 @@
+"""Tests for H3Regionizer."""
+from contextlib import nullcontext as does_not_raise
+from typing import Any, List
+from unittest import TestCase
+
+import geopandas as gpd
+import pytest
+
+from srai.regionizers import S2Regionizer
+
+ut = TestCase()
+S2_RESOLUTION = 7
+
+
+@pytest.fixture  # type: ignore
+def expected_s2_indexes() -> List[str]:
+    """Get expected s2 indexes."""
+    return [
+        "0555c",
+        "0ffe4",
+        "0fff4",
+        "0fffc",
+        "10004",
+        "1000c",
+        "10014",
+        "1001c",
+        "1aaa4",
+        "05554",
+        "1aaac",
+    ]
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "gdf_fixture,expected_s2_indexes_fixture,resolution,expectation",
+    [
+        ("gdf_polygons", "expected_s2_indexes", S2_RESOLUTION, does_not_raise()),
+        ("gdf_multipolygon", "expected_s2_indexes", S2_RESOLUTION, does_not_raise()),
+        ("gdf_empty", "expected_s2_indexes", S2_RESOLUTION, pytest.raises(AttributeError)),
+        ("gdf_polygons", "expected_s2_indexes", -1, pytest.raises(ValueError)),
+        ("gdf_polygons", "expected_s2_indexes", 31, pytest.raises(ValueError)),
+        ("gdf_no_crs", "expected_s2_indexes", S2_RESOLUTION, pytest.raises(ValueError)),
+    ],
+)
+def test_transform(
+    gdf_fixture: str,
+    expected_s2_indexes_fixture: str,
+    resolution: int,
+    expectation: Any,
+    request: Any,
+) -> None:
+    """Test transform of H3Regionizer."""
+    gdf: gpd.GeoDataFrame = request.getfixturevalue(gdf_fixture)
+    s2_indexes: List[str] = request.getfixturevalue(expected_s2_indexes_fixture)
+    with expectation:
+        gdf_s2 = S2Regionizer(resolution).transform(gdf)
+
+        ut.assertCountEqual(first=gdf_s2.index.to_list(), second=s2_indexes)
```

Comment:
Shouldn't you also assert the equality of the indexes? So whether the Regionizer returns the S2 indexes you're actually expecting.

Reply:
Maybe not strict equality, if the order doesn't matter.

Reply:
That's what this method does if I understand it correctly. See [docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual) for reference

Reply:
Yeah, the naming of this function is a bit odd

Reply:
Oh, okay